clang/test/SemaSYCL/sycl-device.cpp

```diff
@@ -34,25 +31,17 @@ class B {
   __attribute__((sycl_device)) virtual void bar() = 0;
 };
 
-#if defined(NOT_STRICT)
-__attribute__((sycl_device))
-int* func3() { return nullptr; }
-
-__attribute__((sycl_device))
-void func3(int *) {}
-#elif defined(WARN_STRICT)
-__attribute__((sycl_device)) // expected-warning {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer return type}}
-int* func3() { return nullptr; }
-
-__attribute__((sycl_device)) // expected-warning {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer parameter type}}
-void func3(int *) {}
-#else
-__attribute__((sycl_device)) // expected-error {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer return type}}
-int* func3() { return nullptr; }
-
-__attribute__((sycl_device)) // expected-error {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer parameter type}}
-void func3(int *) {}
-#endif
+__attribute__((sycl_device)) int *func0() { return nullptr; }
+
+__attribute__((sycl_device)) void func2(int *) {}
+
+__attribute__((sycl_device)) int *func3() { return nullptr; }
+
+__attribute__((sycl_device)) void func4(int *) {}
+
+__attribute__((sycl_device)) int *func5() { return nullptr; }
+
+__attribute__((sycl_device)) void func6(int *) {}
```

Comment:
It seems some of these cases just duplicate each other, can we remove them?
```suggestion
__attribute__((sycl_device)) int *func0() { return nullptr; }

__attribute__((sycl_device)) void func2(int *) {}

```

Reply:
Sorry I overlooked them. Will fix it.